KicadModTree/nodes/Footprint.py

```diff
@@ -57,3 +62,16 @@ def setTags(self, tags):
 
     def setAttribute(self, value):
         self.attribute = value
+
+    def setMaskMargin(self, value):
+        self.maskMargin = value
+
+    def setPasteMargin(self, value):
+        self.pasteMargin = value
+
+    def setPasteMarginRatio(self, value):
+        # paste_margin_ratio is unitless between 0 and 1 while GUI uses percentage
+        if abs(value) > 1:
```

Comment:
input validation using asserts would be cleaner I think:

```python
assert abs(value) <= 1
```

this would also be possible :):

```python
assert -1 <= value <= 1
```